### ISSUE TITLE
Support socket timeout in SysLogHandler to prevent indefinite blocking on socket creation

### ISSUE THREAD

**Issue by hodamarr** [python/cpython#126400]
# Feature or enhancement

### Proposal:

Hi,
In the current implementation of the SysLogHandler class from the logging.handlers module, if the specified address does not respond or is unreachable, the createSocket method may block indefinitely. This is problematic because there is no way to set a socket timeout during the connection's creation without overriding the entire createSocket method.

The only way to add timeout is to override the whole createSocket method (if I use super().createSocket() it will get stuck on this line)

my suggestion:
```python

class SysLogHandler(logging.Handler):
    def __init__(self, address=('localhost', SYSLOG_UDP_PORT), 
                 facility=LOG_USER, socktype=None, timeout=None):
        # Existing initialization
        self.timeout = timeout  # New timeout parameter
        self.createSocket()
    
    def createSocket(self):
        # Existing code ....
        try:
            sock = socket.socket(af, socktype, proto)
            **if self.timeout:
                     sock.setttimeout(self.timeout)**
            # Existing code ........
```
Thanks

### Has this already been discussed elsewhere?

No response given

### Linked PRs
* gh-126716

**Comment by hodamarr** [comment]
Syslog uses both UDP and TCP. A timeout is crucial to avoid blocking the process if the service is unreachable or slow to respond. In my case, I'm using TCP, and the server is unreachable, causing my process to be blocked indefinitely. Adding timeout support to SysLogHandler would allow the application to fail gracefully in such scenarios, instead of hanging due to network issues.

> If I remember correct, the `SysLog` is based on the UDP protocol. I'm sure the timeout is useful for UDP

**Comment by hodamarr** [comment]
> If I remember correct, the `SysLog` is based on the UDP protocol. I'm sure the timeout is useful for UDP

Syslog uses both UDP and TCP. A timeout is crucial to avoid blocking the process if the service is unreachable or slow to respond. In my case, I'm using TCP, and the server is unreachable, causing my process to be blocked indefinitely. Adding timeout support to SysLogHandler would allow the application to fail gracefully in such scenarios, instead of hanging due to network issues.

**Comment by hodamarr** [comment]
Hey @picnixz  I can take care of this—could you assign me to it?

**Comment by picnixz** [comment]
Before starting the implementation, I wonder whether this should count as a bug fix or not (and whether there is another way to fix it). So let's ask @vsajip first.

**Comment by hodamarr** [comment]
Sure, let me know when its okay to open a pull request.
@vsajip @picnixz

**Comment by vsajip** [comment]
Yes, this seems a reasonable feature to add. You're welcome to submit a PR with docs/tests.
